FileDirectoryManager/src/org/jjazz/filedirectorymanager/api/FileDirectoryManager.java

```diff
@@ -149,14 +149,23 @@ public File getNewSongTemplateSongFile()
     public File getJJazzLabUserDirectory()
     {
         // We need a valid user.home directory        
-        String uh = System.getProperty("user.home");
+        final String uh = System.getProperty("user.home");
         assert uh != null : "user.home property does not exist: " + uh;   //NOI18N
         File userHome = new File(uh);
         assert userHome.isDirectory() : "user.home directory does not exist: " + uh;   //NOI18N
 
+        final String dataDir;
+        if (System.getProperty("os.name").equalsIgnoreCase("linux")) {
+            // Follow XDG spec on linux, https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+            final String dataHome = System.getenv("XDG_DATA_HOME");
+            dataDir = (dataHome != null && dataHome.length() > 0) ? dataHome : uh + "/.local/share";
```

Comment:
Are we sure .local/share both exist on all systems ? If not, the mkdir() below should be replaced by mkdirs()

Reply:
Good call. Yes, we should `mkdirs()` the whole path.